### ISSUE TITLE
Unconditional generation

### ISSUE THREAD

**Issue by JCBrouwer** [voletiv/mcvd-pytorch#3]
Hello thanks for the great repository!

I'm wondering how the unconditional video generation results from the paper can be reproduced.
In the runner.video_gen() function there seems to be an assert that only allows conditional generation.
Is there an example for sampling unconditionally elsewhere?

**Comment by JCBrouwer** [comment]
Thanks for the quick response, I managed to get it working I believe.
The results were all very gray at first, but started to look better after they had drifted for enough frames.
Are there any settings during training that might help improve the performance specifically on unconditional generation?

Also is it possible to run video_gen() and only do the unconditional generation step?
I tried to puzzle out all the if statements but was just left confused.
I added an uncond_gen() function to the runner but that just ended up deleting my log folder, there goes a week and a half of training ;_;

**Comment by AlexiaJM** [comment]
This grey effect seems very weird, id make sure that your data has the correct format [-1,1] and that you use sampling.clip_before=True and model.version=DDPM during sampling.

The easy hacky way to do it would be to do use data.num_frames_cond=1 with data.prob_mask_cond=1 during training, this way it always mask the single conditional frame and its always just unconditional. vid-gen will still do video prediction conditional, but you can just ignore it. If you want to remove conditional generation, here's a suggestion: In https://github.com/voletiv/mcvd-pytorch/blob/master/runners/ncsn_runner.py, remove lines 1508 to 1571 and then change the if statement in line 1573 so it activates this one to prevent fvd/ssim/etc calculation.




.

**Comment by YingqingHe** [comment]
> 

hi, thanks for your great work.

**Comment by YingqingHe** [comment]
Hi! Thanks for your great work. 

I am wondering what is the exact data resolution of the unconditional generation on UCF-101? 

I have seen the paper report the FVD metrics in table 6 (Unconditional generation of UCF-101 video frames.) in the main paper, however, I am confused about the data resolution.
